rasa/shared/utils/io.py

```diff
@@ -518,26 +521,56 @@ def raise_deprecation_warning(
     raise_warning(message, FutureWarning, docs, **kwargs)
 
 
+def read_validated_yaml(filename: Union[Text, Path], schema: Text) -> Any:
+    """Validates YAML file content and returns parsed content.
+
+    Args:
+        filename: The name of the file to read.
+        schema: The path to the schema file which should be used for validating the
+            file content.
+
+    Returns:
+        The parsed file content.
+
+    Raises:
+        YamlValidationException: In case the model configuration doesn't match the
+            expected schema.
+    """
+    content = read_file(filename)
+
+    rasa.shared.utils.validation.validate_yaml_schema(content, schema)
+    return read_yaml(content)
+
+
 def read_config_file(filename: Union[Path, Text]) -> Dict[Text, Any]:
     """Parses a yaml configuration file. Content needs to be a dictionary
 
     Args:
         filename: The path to the file which should be read.
+
+    Raises:
+        YamlValidationException: In case file content is not a `Dict`.
+
+    Returns:
+        Parsed config file.
     """
-    content = read_yaml_file(filename)
+    return read_validated_yaml(filename, CONFIG_SCHEMA_FILE)
 
-    if content is None:
-        return {}
-    elif isinstance(content, dict):
-        return content
-    else:
-        raise YamlSyntaxException(
-            filename,
-            ValueError(
-                f"Tried to load configuration file '{filename}'. "
-                f"Expected a key value mapping but found a {type(content).__name__}"
-            ),
-        )
+
+def read_model_configuration(filename: Union[Path, Text]) -> Dict[Text, Any]:
+    """Parses a model configuration file.
+
+    Args:
+        filename: The path to the file which should be read.
+
+    Raises:
+        YamlValidationException: In case the model configuration doesn't match the
```

Comment:
that's a different one than before but in my opinion the correct one. Double checked the code and nothing depends on the exact exception type.